compiler/rustc_hir_typeck/src/pat.rs

```diff
@@ -1966,6 +1971,42 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         box_ty
     }
 
+    fn check_pat_deref(
+        &self,
+        span: Span,
+        inner: &'tcx Pat<'tcx>,
+        expected: Ty<'tcx>,
+        pat_info: PatInfo<'tcx, '_>,
+    ) -> Ty<'tcx> {
+        let tcx = self.tcx;
+        // FIXME(deref_patterns): use `DerefPure` for soundness
+        // FIXME(deref_patterns): use `DerefMut` when required
+        // <expected as Deref>::Target
+        let ty = Ty::new_projection(
+            tcx,
+            tcx.require_lang_item(hir::LangItem::DerefTarget, Some(span)),
+            [expected],
+        );
+
+        let mut fulfill_cx = <dyn TraitEngine<'_>>::new(&self.infcx);
```

Comment:
it's unnecessary to make a new fulfillment context

just use `.normalize` then `.try_structurally_resolve`

Reply:
Like this: [`7400f95` (#122222)](https://github.com/rust-lang/rust/pull/122222/commits/7400f958d8d9f1684f883a873c33fd9f4a1b0d21)?